### ISSUE TITLE
Fine Tuning for Arabic

### ISSUE THREAD

**Issue by MrMuhannadObeidat** [canopyai/Orpheus-TTS#110]
I am fine tuning the [orpheus-3b-0.1-ft](https://huggingface.co/canopylabs/orpheus-3b-0.1-ft) for Arabic languague. I have around 6500 audio samples with varying lengths.
I tried multiple training recipes:
1. Full model training with batch size of 1 and gradient accumulation of 4. 10 epochs will drive the loss significantly down, very close to zero.
2. Unsloth based approach documented by Orpheus in their notebooks.
3. Lora based approach.
In all cases I am using SNAC 24K model and reducing my sampling rate of the audio from 48K.

Now for sentences similar to the training data, the generation is good and for the most part easy to understand but not 100% perfect. I am using a speaker tag so it uses that same style rather than a generic voice. The speed of speech is high and not sure if that is something I can control with any additional parameters.

For any net new sentences the model is not able to speak in intelligible way. So here are my questions:
1. I know I have to fix my training to avoid loss going down to zero so working on that.
2. Do you think 6K samples is enough or too low for this goal?
3. Most importantly, do I need to build an extended tokenizer for Arabic phonemes. Is it at all possible to train a new language such as Arabic without touching the tokenizer?

**Comment by MrMuhannadObeidat** [comment]
Thanks for the prompt response and quick suggestions. I have a few of follow ups here:
1. For tokenizer, I meant to refer to the audio part coming up with phonemes rather than the text tokens. What do you think about that?
2. For the text part, I saw an earlier post by you suggesting we convert non-english to english during data set prep phase. "if you can programmatically transliterate phonetically into roman alphabet then these would probably reduce the training burden" https://github.com/canopyai/Orpheus-TTS/issues/10#issuecomment-2738534911
3. For the two stage training, and since I have about 8 hours of voice data, I tried to do one run without specifying a speaker than using Tara for inference. That did not produce good output. Later on I used all the data I have to retrain but with speaker specified and that improved inference a bit. For the approach you mentioned, do you have any documentation on it.

**Comment by kadirnar** [comment]
> Thanks for the prompt response and suggestions. I have a few of follow ups here:
> 
> 1. For tokenizer, I meant to refer to the audio part coming up with phonemes rather than the text tokens. What do you think about that?
> 2. For the text part, I saw an earlier post by you suggesting we convert non-english to english during data set prep phase. "if you can programmatically transliterate phonetically into roman alphabet then these would probably reduce the training burden" [training / finetuning other languages #10 (comment)](https://github.com/canopyai/Orpheus-TTS/issues/10#issuecomment-2738534911)
> 3. For the two stage training, and since I have about 8 hours of voice data, I tried to do one run without specifying a speaker than using Tara for inference. That did not produce good output. Later on I used all the data I have to retrain but with speaker specified and that improved inference a bit. For the approach you mentioned, do you have any documentation on it.

Which dataset are you using? I'm testing this dataset too. However, I need a higher quality dataset.

**Comment by amuvarma13** [comment]
1. You don't need to modify the snac tokeniser either in my opinion
2. I think you should use this as a backup plan
3. You should really be training the pretrained model not the finetuned one - none of the English voices will work with Arabic

We will be releasing a guide shortly as well as some multilingual models that should help people with their own stuf.

**Comment by dsinghrathore** [comment]
Hey @amuvarma13, the readme.md mentions "If you are doing extended training this model, i.e. for another language or style we recommend starting with finetuning only (no text dataset)"

**Comment by kadirnar** [comment]
@amuvarma13 I trained it with a Quran Arabic dataset but it can't read the words correctly. Could it be related to the tokenizer? However, the Llama3 model understands Arabic.

Dataset: https://huggingface.co/datasets/MohamedRashad/Quran-Recitations

https://voca.ro/11qLflvlJcuM

**Comment by kadirnar** [comment]
I will train with this dataset. This dataset is about daily conversation.

https://huggingface.co/datasets/ymoslem/MediaSpeech

**Comment by kadirnar** [comment]
> I will train with this dataset. This dataset is about daily conversation.
> 
> https://huggingface.co/datasets/ymoslem/MediaSpeech

https://voca.ro/175jW5CbnUWh
https://voca.ro/1f1ockUrdI70

**Comment by amuvarma13** [comment]
Move to discussion and close issue?

**Comment by thewh1teagle** [comment]
Hi everyone,
I would like to fine tune the English model to new language. 
I created accurate phonemizer for that and I have the dataset with phonemes. 
Does Orpheus trained on phonemes? I hope to leverage it when fine tuning, so I'll fine tune on similar IDS. 
If I train on phonemes, will it leverage the pre trained model?

**Comment by kadirnar** [comment]
@blueyo 
The MediaSpeech-AR dataset is very small. We need a larger dataset. Right now I'm focusing on Japanese and Chinese languages.

**Comment by blueyo** [comment]
@kadirnar I can work on the preparing dataset for Arabic for the Orpheus model, could you please share guide on how I should fineturn for Arabic? I am new to TTS in general, but I do have time and resources to work on Arabic language your help and guidance

**Comment by kadirnar** [comment]
> [@kadirnar](https://github.com/kadirnar) I can work on the preparing dataset for Arabic for the Orpheus model, could you please share guide on how I should fineturn for Arabic? I am new to TTS in general, but I do have time and resources to work on Arabic language your help and guidance

You can prepare it by looking at the sample dataset.
https://colab.research.google.com/github/unslothai/notebooks/blob/main/nb/Orpheus_(3B)-TTS.ipynb

**Comment by mohamednaji7** [comment]
> The MediaSpeech-AR dataset is very small. We need a larger dataset. Right now I'm focusing on Japanese and Chinese languages.

@kadirnar  maybe YouTube ASR transcripts would be good for pre-training
